common/optee/optee.edl

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+/*
+**==============================================================================
+**
+** optee.edl:
+**
+**     This file declares internal ECALLs/OCALLs used by liboehost/liboecore
+**     that are specific to OP-TEE.
+**
+**==============================================================================
+*/
+
+enclave
+{
+    include "openenclave/bits/types.h"
+
+    trusted
+    {
+        public void oe_optee_stub_ecall();
+    };
+
+    untrusted
+    {
+        void* oe_calloc_ocall(
```

Comment:
Does this function returns a pointer (to host heap) that cannot be used inside enclave? If so, what's the point?

To be clear, I was actually expecting this call to return a pointer to the shared memory pool on TrustZone.

Reply:
The returned pointer points to host-side memory. The issue is that this function is public, so it's expected to be implemented, even if it's useless on TrustZone. This is the reason why I argued for removing it from the public API surface altogether.

The shared memory pool (or dynamically shared memory buffers, which temporarily map host memory into the enclave) is only used in passing to marshal ECALLs/OCALLs, not for any storage longer than a function call boundary.